### DISCUSSION TITLE
Using a single Body parameter, the JSON format will report an error

### DISCUSSION THREAD

**Discussion by wxq0309** [fastapi/fastapi#7640] [Questions] [answered]
### Description
Using a single Body parameter, the JSON format will report an error

### Additional context
```
@router.post("/test/")
async def test(abd: str=Body(...)):
    print(abd)               ---error

@router.post("/test/")
async def test(abd: str=Body(...), acd: str=Body(...)):
    print(abd)               ---right

Using a single Body parameter, the JSON format will report an error

**Comment by phy25** [comment]
What JSON format is producing an error? What is the error message? Could you provide an example request?

**Comment by wxq0309** [comment]
> What JSON format is producing an error? What is the error message? Could you provide an example request?

I want to send a JSON data in the format of  {" abd ":" message "}, but when I use a single Body parameter setting, send it in the above JSON format, it will prompt
```
{
  "detail": [
    {
      "loc": [
        "body",
        "abd"
      ],
      "msg": "str type expected",
      "type": "type_error.str"
    }
  ]
}

**Comment by kajogo777** [comment]
@wxq0309 how did you resolve this? I am having the same issue

**Comment by wxq0309** [comment]
> @ wxq0309您如何解决此问题？我有相同的问题

This can be done by setting the Embed property

```
item: Item = Body(..., embed=True)

example:
from fastapi import Body, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str = None
    price: float
    tax: float = None


@app.put("/items/{item_id}")
async def update_item(*, item_id: int, item: Item = Body(..., embed=True)):
    results = {"item_id": item_id, "item": item}
    return results
```

**Comment by kajogo777** [comment]
thank you

**Comment by tiangolo** [comment]
Thanks for the discussion here everyone! And thanks @wxq0309 for closing the issue and reporting back your solution. :+1:

**Comment by BillGrim** [comment]
> > @ wxq0309您如何解决此问题？我有相同的问题
> 
> This can be done by setting the Embed property
> 
> ```
> item: Item = Body(..., embed=True)
> 
> example:
> from fastapi import Body, FastAPI
> from pydantic import BaseModel
> 
> app = FastAPI()
> 
> 
> class Item(BaseModel):
>     name: str
>     description: str = None
>     price: float
>     tax: float = None
> 
> 
> @app.put("/items/{item_id}")
> async def update_item(*, item_id: int, item: Item = Body(..., embed=True)):
>     results = {"item_id": item_id, "item": item}
>     return results
> ```

In my opinion, the condition of your solution is already different from the problem you met, which is "Single parameter declared with fastapi.Body requests str formatted rather than json formatted request body".

I read the official doc [singular-values-in-body](https://fastapi.tiangolo.com/tutorial/body-multiple-params/#singular-values-in-body), but the demo contains extra paramters(request param, pydantic model).

In my case, I have many apis that needs only a single post body param.

So Is there a way to declare a single param in body of  a post api without using pydantic?

**Comment by rafitc** [comment]
I'm encountering some strange behavior in my FastAPI application. I'm sending a single body parameter to a route handler. When I use a synchronous database connection dependency injection, everything works fine, and I don't get any validation errors. (without embed=true)

However, when I switch to using an asynccontext for the database connection manager, I start getting a ValidationError for the same body parameter. The body parameter is defined using `[Body(alias="...")]`  and the request payload is correctly formatted. It's getting resolved if i add `embed=True`.

I can't figure out why this difference in behavior is happening. Has anyone else experienced this issue, or does anyone have insights into what might be causing this?

**Reply by YuriiMotov** [reply]
Could you provide a [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example)?

**Reply by YuriiMotov** [reply]
@rafitc , I closed this discussion as answered. If the answer doesn't work for you, please open new discussion and provide MRE. Will be glad to help you then!

**Reply by rafitc** [reply]
I'll provide the MRE shortly.

**Comment by YuriiMotov** [comment, marked as answer]
Answering the initial question, `embed=True` is a key here.

The following code works fine (at least in current version of FastAPI):

```
from fastapi import APIRouter, Body, FastAPI

router = APIRouter()

@router.post("/test/")
async def test(abd: str=Body(..., embed=True)):
    print(abd)

app = FastAPI()

app.include_router(router)

```

Tested with:

```
curl -X 'POST' \
  'http://localhost:8000/test/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{"abd": "string"}'
```

By passing `embed=True` you force FastAPI to include this parameter as **Key-Value** pair instead of just **Value**.
Without `embed=True` FastAPI will expect body to be just `str`. So, to pass validation, the request should be sent like this:
```
curl -X 'POST' \
  'http://localhost:8000/test/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d 'string'
```

**Reply by rafitc** [reply]
I agree that this will work. 

However, I have a very large codebase that was working without the `embed=True` option initially. After changing the database dependency manager to use async DB pooling, this issue suddenly started to pop up. I'm not finding any logical reason for this, as the request body validation is breaking after changing the database connection mode.
